golem-component-service/src/authed/component.rs

```diff
@@ -0,0 +1,540 @@
+// Copyright 2024-2025 Golem Cloud
+//
+// Licensed under the Golem Source License v1.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://license.golem.cloud/LICENSE
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::error::ComponentError;
+use crate::model::InitialComponentFilesArchiveAndPermissions;
+use crate::model::{Component, ComponentByNameAndVersion, ComponentConstraints};
+use crate::service::component::ComponentService;
+use bytes::Bytes;
+use futures_util::stream::BoxStream;
+use golem_common::model::auth::AuthCtx;
+use golem_common::model::auth::ProjectAction;
+use golem_common::model::component::ComponentOwner;
+use golem_common::model::component::VersionedComponentId;
+use golem_common::model::component_constraint::FunctionConstraints;
+use golem_common::model::component_metadata::DynamicLinkedInstance;
+use golem_common::model::plugin::PluginOwner;
+use golem_common::model::plugin::{
+    PluginInstallation, PluginInstallationAction, PluginInstallationCreation,
+    PluginInstallationUpdate,
+};
+use golem_common::model::{ComponentId, ComponentType, ComponentVersion, PluginInstallationId};
+use golem_common::model::{InitialComponentFile, ProjectId};
+use golem_service_base::clients::auth::BaseAuthService;
+use golem_service_base::clients::project::ProjectService;
+use golem_service_base::model::ComponentName;
+use std::collections::HashMap;
+use std::fmt::{Debug, Formatter};
+use std::sync::Arc;
+use std::vec;
+
+pub struct AuthedComponentService {
+    base_component_service: Arc<dyn ComponentService>,
+    auth_service: Arc<dyn BaseAuthService>,
+    project_service: Arc<dyn ProjectService>,
+}
+
+impl AuthedComponentService {
```

Comment:
Getting rid of this separate layer of services with auth handling is not really possible without making everything more complex. Problem is that there are a lot of interdependencies between the services, so keeping this separate layer and doing auth at the edge makes things simples.